File: 20200219/carpet_test.py
def solution(brown, red):
    [[x, y]] = [i for i in comb(int((brown - 4) / 2)) if i[0] * i[1] == red]
    return [y + 2, x + 2]


def comb(num):
    return [[i, num - i] for i in range(1, num // 2 + 1)]


def test_solution():
    assert solution(10, 2) == [4, 3]


def test_comb():
    assert comb(3) == [[1, 2]]

File: 20200221/carpet.test.js
const solution = (brown, red) => {
  let w = 0;
  let h = 0;

  do {
    h += 1;
    w = red / h;
  } while (brown !== 2 * (w + h + 2))

  return [w + 2, h + 2];
}

test('solution', () => {
  expect(solution(10, 2)).toEqual([4, 3]);
  expect(solution(8, 1)).toEqual([3, 3]);
  expect(solution(24, 24)).toEqual([8, 6]);
});
